cmd/oc-mirror/main.go

```diff
@@ -15,3 +40,48 @@ func checkErr(err error) {
 		kcmdutil.CheckErr(err)
 	}
 }
+
+func isV2() bool {
+	return len(os.Args) > 0 && slices.Contains(os.Args, "--v2")
+}
+
+func runOcMirrorV2Cmd(args []string) error {
+	tmpdir, err := os.MkdirTemp("", "oc-mirror-")
+	if err != nil {
+		return fmt.Errorf("failed to create tmpdir: %w", err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	path := filepath.Join(tmpdir, "oc-mirror")
+	klog.V(5).Infof("Unpacking v2 binary to %s", path)
+
+	binaryV2, err := mirrorv2.ReadFile("data/oc-mirror-v2")
+	if err != nil {
+		return fmt.Errorf("failed to read v2 binary: %w", err)
+	}
+
+	v2File, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create v2 binary: %w", err)
+	}
+	if _, err := io.Copy(v2File, bytes.NewReader(binaryV2)); err != nil {
+		v2File.Close()
+		return fmt.Errorf("failed to write v2 binary: %w", err)
+	}
+	// We must close the binary file before we can execute it
+	v2File.Close()
+
+	cmd := exec.Cmd{
+		Path: path,
+		SysProcAttr: &syscall.SysProcAttr{
+			// Kill children if parent is dead
+			Pdeathsig: syscall.SIGKILL,
+			Setpgid:   true,
+		},
+		Args:   args,
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+	}
+
+	return cmd.Run()
```

Comment:
Hi @r4f4 
this is an interesting way to call the v2. 
I'm not at all an expert... I wonder how this impacts the way oc-mirror would run in the various situations:
* as a standalone binary
* as an oc plugin
* inside a pod / pipeline

maybe not at all...

Reply:
There should be no impact I know of. If there is, it's probably a bug in my implementation.

Reply:
I was also thinking about impacts on the debugger (dev only).
When we previously tried to use buildah as the solution to build catalogs, there was also an exec.Cmd involved, and debuggings were harder (but not impossible) to do

Reply:
> I was also thinking about impacts on the debugger (dev only).
> When we previously tried to use buildah as the solution to build catalogs, there was also an exec.Cmd involved, and debuggings were harder (but not impossible) to do

Then you'll be happy to know this has no impact! For local dev/debug you can run the V2 binary directly. The run-v2-from-v1 is just a convenience for users, devs don't have to go through the wrapper.